src/Contract/Operation/Zipable.php

```diff
@@ -22,9 +22,12 @@ interface Zipable
      *
      * @see https://loophp-collection.readthedocs.io/en/stable/pages/api.html#zip
      *
-     * @param iterable<mixed, mixed> ...$iterables
+     * @template U
+     * @template UKey
      *
-     * @return Collection<list<mixed>, list<mixed>>
+     * @param iterable<UKey, U> ...$iterables
```

Comment:
I know that it's not possible to type this thing as it is a variadic parameter.

I don't know if it's a good idea, but I had the idea to do the typing of only one element, knowing that it would not be completely valid if there are more than one parameters given.

WDYT?

Reply:
I get what you're looking to do here. The only issue is that when a user will try to use it with multiple iterables of different types they will get an error and so think that it's not supposed to be used like that.

I'm not sure how people normally use `zip`, maybe it wouldn't be an issue if most often it's used with a single type.

Reply:
I leave this question open as I don't have any idea yet on what is best to do...